proto/sifnode/ethbridge/v1/tx.proto

```diff
@@ -125,4 +126,14 @@ message MsgSetFeeInfo {
   ];
 }
 
-message MsgSetFeeInfoResponse {}
\ No newline at end of file
+message MsgSetFeeInfoResponse {}
+
+// MsgSignProphecy defines a message for sending signature for prophecy
+message MsgSignProphecy {
```

Comment:
@juniuszhou This data structure is incorrect. In order for this to work, the witness must sign off on the following data:

struct ClaimData {
        bytes cosmosSender;
        uint256 cosmosSenderSequence;
        address payable ethereumReceiver;
        address tokenAddress;
        uint256 amount;
        bool doublePeg;
        uint128 nonce;
}

The data should be concatenated using the abi.encode method before being signed. The data should be signed the order specified above. This data that needs to be signed off on can be viewed in the cosmos bridge contract on the future/peggy2 branch starting at line 155 where the struct is defined.

Reply:
This message is used for relayer to send the signature to sifnode after find out the event of new prophecy. The prophecy id is the hash of claimData in my implementation. so relayer just sign off on the prophecy id.